doc/source/user/tutorial-ma.rst

```diff
@@ -0,0 +1,377 @@
+=======================
+Tutorial: Masked Arrays
+=======================
+
+.. currentmodule:: numpy
+
+.. testsetup::
+
+   import numpy as np
+   np.random.seed(1)
+
+**Prerequisites**
+
+Before reading this tutorial, you should know a bit of Python. If you
+would like to refresh your memory, take a look at the
+:doc:`Python tutorial <python:tutorial/index>`.
+
+If you want to be able to run the examples in this tutorial, you should also
+have `matplotlib <https://matplotlib.org/>`_ installed on your computer.
+
+**Learner profile**
+
+This tutorial is for people who have a basic understanding of NumPy and want to
+understand how masked arrays and the :mod:`numpy.ma` module can be used in
+practice.
+
+**Learning Objectives**
+
+After this tutorial, you should be able to:
+
+- Understand what are masked arrays and how they can be created
+- Understand how to access and modify data for masked arrays
+- Decide when the use of masked arrays is appropriate in some of your
+  applications
+
+**What are masked arrays?**
+
+Consider the following problem. You have a dataset with missing or invalid
+entries. If you're doing any kind of processing on this data, and want to
+`skip` or flag these unwanted entries without just deleting them, you may have
+to use conditionals or filter your data somehow. The :mod:`numpy.ma` module
+provides some of the same funcionality of
+:class:`NumPy ndarrays <numpy.ndarray>` with added structure to ensure
+invalid entries are not used in computation.
+
+From the :mod:`Reference Guide <numpy.ma>`:
+
+    A masked array is the combination of a standard :class:`numpy.ndarray` and
+    a **mask**. A mask is either ``nomask``, indicating that no value of the
+    associated array is invalid, or an array of booleans that determines for
+    each element of the associated array whether the value is valid or not.
+    When an element of the mask is ``False``, the corresponding element of the
+    associated array is valid and is said to be unmasked. When an element of
+    the mask is ``True``, the corresponding element of the associated array is
+    said to be masked (invalid).
+    
+
+We can think of a :class:`MaskedArray <numpy.ma.MaskedArray>` as a
+combination of:
+
+- Data, as a regular :class:`numpy.ndarray` of any shape or datatype;
+- A boolean mask with the same shape as the data;
+- A ``fill_value``, a value that may be used to replace the invalid entries
+  in order to return a standard :class:`numpy.ndarray`.
+
+**When can they be useful?**
```

Comment:
I think you should make these all rST headings so that they can be linked to.

That would look something like:

```suggestion
When can they be useful?
========================
```

Reply:
This was a point made in my other tutorial (tutorial-svd) where I had done all headings and it was requested that I remove them because they were all showing in a global toc somewhere.